src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -25,6 +25,9 @@ public class ViewBuffer : IHtmlContentBuilder
         private readonly IViewBufferScope _bufferScope;
         private readonly string _name;
         private readonly int _pageSize;
+        private ViewBufferPage _singlePage = null;          // Limits allocation if the ViewBuffer has only one page (frequent case).
```

Comment:
@Yves57, yeah wouldn't hurt to simplify `Append_CreatesManyPages`. Consider refactoring the 2nd set of Act and Assert (`Assert_CreatesSinglePage`) into a separate test if that hasn't already been covered.